### DISCUSSION TITLE
Migrate Firebase Token to Service Account for Firebase App Distribution

### DISCUSSION THREAD

**Discussion by hoangnguyen92dn** [nimblehq/android-templates#328] [RFCs]
## Issue

Firebase announced new updates about deprecating Firebase Token would be removed in the future major version. It's time to migrate to using Service Account

![image](https://user-images.githubusercontent.com/6950766/197436831-f4a34344-2194-4942-9873-9e1be77571f4.png)

## Solution

Since we are using [wzieba/Firebase-Distribution-Github-Action](https://github.com/wzieba/Firebase-Distribution-Github-Action), there is a guideline for migration from the author. We could follow up to update our `.cicdtemplate` ([.github actions](https://github.com/nimblehq/android-templates/blob/develop/.cicdtemplate/.github/workflows/deploy_staging_and_production_to_firebase_app_distribution.yml))

Reference: https://github.com/wzieba/Firebase-Distribution-Github-Action/wiki/FIREBASE_TOKEN-migration
 
## Who Benefits?

Android chapter
 
## What's Next?

- [ ] Update the CICD template
- [ ] Add README.md in `.cicdtemplate` to show steps how to generate the credentials for usage

**Comment by toby-thanathip** [comment]
@hoangnguyen92dn Yes, we should definitely migrate at some point 👍 

> ⚠️ Deprecated! Don't use it. Firebase team deprecated this option and it will soon be removed.

Any idea when this will be? I'm having a hard time finding the answer 💭

**Reply by toby-thanathip** [reply]
> This means that actions configured with the token parameter will stop working as soon as a new major version of firebase-tools is released.

Quite vague at the moment on what's planned for v12: https://github.com/firebase/firebase-tools/milestones 🙈 

![Screenshot 2022-10-24 at 15 23 26](https://user-images.githubusercontent.com/18277915/197481428-8906bb81-4c83-4034-aacb-9abfa8bc8063.png)

**Comment by luongvo** [comment]
@Tuubz @hoangnguyen92dn FYI, it's stopped working from my KMM IC now https://github.com/luongvo/kmm-survey/actions/runs/3582764458/jobs/6027483936. I think it's time for the migration 👮 

![image](https://user-images.githubusercontent.com/16315358/204959840-d7fac68f-851c-4407-8fef-96696ac514de.png)

**Comment by luongvo** [comment]
@Tuubz I think we should request for voting on this RFC instead, then you can finalize the content on https://github.com/nimblehq/android-templates/issues/360 🙏

**Reply by toby-thanathip** [reply]
Voted ✅

**Comment by luongvo** [comment]
@hoangnguyen92dn @ryan-conway Please speed up the voting process and prepare tickets for this RFC soon 🙏

**Comment by chornerman** [comment]
Voted 🚀

**Comment by ryan-conway** [comment]
[Issue](https://github.com/nimblehq/android-templates/issues/445) created 🙏🏻
